tests/testthat/test-metareactive.R

```diff
@@ -37,12 +37,93 @@ test_that("doesn't break metaprogramming with quosures", {
 
     expect_identical(result1, "ok")
 
-    expect_snapshot_output(formatCode(withMetaMode(o1())))
+    expect_snapshot_output(withMetaMode(o1()))
 
     # Outputs
 
     out1 <- rlang::inject(metaRender(shiny::renderText, !!outer_quo))
     expect_identical(out1(), "ok")
+  })
+})
+
+describe("metaAction", {
+  it("basically works", {
+    a <- 1
+    metaAction({
+      a <- 2
+    })
+    expect_identical(a, 2)
+
+    metaAction(quote({
+      a <- 3
+    }), quoted = TRUE)
+    expect_identical(a, 3)
+
+    env <- new.env()
+    metaAction({
+      a <- 4
+    }, env = env)
+    expect_identical(a, 3)
+    expect_identical(env[["a"]], 4)
+
+    metaAction(quote({
+      a <- 5
+    }), env = env, quoted = TRUE)
+    expect_identical(a, 3)
+    expect_identical(env[["a"]], 5)
+  })
+
+  it("unquotes properly", {
+    b <- TRUE
+    act <- metaAction(x <- ..(b))
```

Comment:
Do we care enough to also test `!!` usage?

Reply:
That's a good idea--we have enough to context to write a test now, we might lose it by the next time this comes up